dotnet/src/Microsoft.Agents.AI.AGUI/AGUIAgentThread.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Text.Json;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Agents.AI.AGUI;
+
+internal sealed class AGUIAgentThread : InMemoryAgentThread
```

Comment:
why is `AGUIAgentThread` based on `InMemoryAgentThread` particularly? Can't it be base `AgentThread`? I am thinking we should not restrict type as much beforehand